lib/models/user.rb

```diff
@@ -0,0 +1,30 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String, unique: true, required: true
+  property :username, String, unique: true, required: true
+  property :password_digest, Text
+  property :name, String
+
+  has n, :peeps
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+
+  def self.authenticate username, password
+    user = User.first(username: username)
+    if user && BCrypt::Password.new(user.password_digest) == password
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.